addons/mail/static/src/components/composer/composer.js

```diff
@@ -48,11 +48,24 @@ class Composer extends Component {
          * Reference of the text input component.
          */
         this._textInputRef = useRef('textInput');
+        /**
+         * Reference of the subject input. Useful to set content.
+         */
+        this._subjectRef = useRef('subject');
```

Comment:
Can't you just use `t-att-value` on the template instead of all of this? It's not possible for textarea due to textarea being strange, but it should work for text input?

Reply:
It works for everything but the reset to 0, so I left the patched/mounted things

Reply:
What is the value when resetting to 0? t-att has some conditions to not render the attribute at all when the value is falsy that should be adapted or worked around probably, so that we get an empty value instead.

Reply:
It contains the value before the post message. On post message we set the field to `""` but it's not applied in template

Reply:
This really makes the business code looks like a mess though.

Maybe we could create a hook for that? And also one for the textarea?

Goal of the hook being to properly intercept events and execute a given user function (in this case, update model) and also automatically set value on mout/patch, again with a given function (in this case, returning model value).

Pretty much a generic version of t-model https://github.com/odoo/owl/blob/master/doc/reference/component.md#form-input-bindings that can work with other things than state.

Reply:
how would you name it ? do you think it should be done in this PR ?

Reply:
I'm not sure, what do you think @alexkuhn?

This looks out of scope, but then it should actually be done as a prerequisite of this PR and not after.

Did you remove all the event handlers? How is the value written on the record now?

Reply:
`t-value` works fine for all events except putting it back to `""` when message is posted, so basically yeah I removed everything except the calls in `patched()` and `mounted()`

Reply:
Wait what, `t-value`? That's just supposed to be used to define a new variable name inside the template. I'm not even sure how that works and it isn't suppose to.

What I mean is `t-att-value` which should write the new value (even if empty) correctly then?
But the events should be handled manually again.

The only way with OWL to do what we want here with form elements is using `t-model` except it doesn't work for anything else than a direct read/assignation into a (proxified) state.

Reply:
Think it lacks of a test because it wasn't working but tests were passing

Reply:
(Added the test)

Reply:
> I'm not sure, what do you think @alexkuhn?
> 
> This looks out of scope, but then it should actually be done as a prerequisite of this PR and not after.

It was acceptable to put all these stuffs in composer text input when just itself need syncing value with models, but now that there's another use with subject, I would definitely find a way to define it once and make it work in both inputs.

I don't think it's a component hook we need. Kinda like how we had a hook for file uploader, and in the end we defined a component `FileUploader`, I think this is also a component here. The purpose of this component is to display an input (broad term, could be input/textarea/contenteditable, or even work with slots?) and have simple API to intercept changes of content and to update its content.

Reply:
Maybe we could merge this like that and add a task to refine this part of code (even if it implies I have to do it just right after this is merged) ? This commit fixes 3 different bugs and could be useful for other PR's like https://github.com/odoo/odoo/pull/53428 for the border-radius variables introduced.

I think we need to fix all bugs we can (without introducing too much technical debt off course).
This refactoring is needed and makes sense but it can take a little while to make it work correctly without any bug fixing included which is our priority for now, if I understood well (and even leaving 3 bugs there that could have been fixed without him)

Reply:
Agree to not delay this PR with such drastic changes. Let's plan this for after merge.

Reply:
Ok too, please create task and add appropriate fixme

Reply:
task-2284290